src/node.cc

```diff
@@ -3848,6 +3854,9 @@ static void ParseArgs(int* argc,
     } else if (strcmp(arg, "--expose-http2") == 0 ||
                strcmp(arg, "--expose_http2") == 0) {
       // Keep as a non-op through v9.x
+    } else if (strcmp(arg, "--preserve-dns-order") == 0 ||
+               strcmp(arg, "--preserve_dns_order") == 0) {
+      config_preserve_dns_order = true;
```

Comment:
Most Node.js command line options only support the `-` variants (i.e. `--preserve-dns-order`). As this stands the other variant, `--preserve_dns_order`, would be allowed on the command line but not in `NODE_OPTIONS`.